src/app/app-directives/course-status.directive.ts

```diff
@@ -0,0 +1,36 @@
+import { Directive, ElementRef, Input, Renderer2, OnInit } from '@angular/core';
+import { todayWithoutTime } from '@app/app-utils/utils';
+
+/** Styles course border depending on its creation date. */
+@Directive({
+  selector: '[appCourseStatus]'
+})
+export class CourseStatusDirective implements OnInit {
+  @Input('appCourseStatus') courseCreationDate: Date;
+
+  constructor(private element: ElementRef, private renderer: Renderer2) {
+  }
+
+  ngOnInit() {
+    this.calculateBorderStyle();
+  }
+
+  private calculateBorderStyle() {
+    let borderClass = null;
+    const today = todayWithoutTime();
+    const prev14days = todayWithoutTime();
+    prev14days.setDate(today.getDate() - 14);
+
+    const todayTimestamp = today.getTime();
+    const prev14daysTimestamp = prev14days.getTime();
+    const courseCreationTimestamp = this.courseCreationDate.getTime();
+
+    if (courseCreationTimestamp < todayTimestamp && courseCreationTimestamp >= prev14daysTimestamp) {
```

Comment:
Usually, I recommend to extract the many conditions in the `if` clause to a variable, let's call it `isFresh`. Then the `if` turns way more clearer.

What do you think?

Reply:
Sounds good - I'll refactor this in upcoming task!